app/pages/tradingNames/DeleteAllTradingNamesPage.scala

```diff
@@ -33,6 +35,10 @@ case object DeleteAllTradingNamesPage extends QuestionPage[Boolean] {
 
   override protected def nextPageCheckMode(waypoints: NonEmptyWaypoints, answers: UserAnswers): Page =
     answers.get(this) match {
+      case Some(_) if waypoints.inAmend =>
+        waypoints.waypoints.toList.collectFirst( waypointPage => waypointPage.page).getOrElse{
+          throw new IllegalStateException("Error retrieving the change registration page")
```

Comment:
Alternative approach to this is to include the `ChangeRegistrationPage.urlFragment` in the `fragments` variable inside the `Waypoint` object. So, the updated variable should look something like this:
 ```
private val fragments: Map[String, Waypoint] =
    Map(
      ... -> ...
      ChangeRegistrationPage.urlFragment -> ChangeRegistrationPage.waypoint,
      ... -> ...
    )
```

This should hopefully work!